api/src/main/java/org/eclipse/microprofile/rest/client/RestClientBuilder.java

```diff
@@ -17,13 +17,29 @@
 package org.eclipse.microprofile.rest.client;
 
 import javax.annotation.Priority;
+import javax.ws.rs.core.Configurable;
 import java.net.URL;
 import java.util.ArrayList;
 import java.util.Comparator;
 import java.util.List;
 import java.util.ServiceLoader;
+import java.util.function.ToIntFunction;
 
-public abstract class RestClientBuilder {
+/**
+ * <p>
+ * This is the main entry point for creating a Type Safe Rest Client.  Implementations are expected
+ * to implement this class and register a service provider via {@link ServiceLoader} that works within their implementation.
+ * <p>
+ * Invoking <pre>RestClientBuilder.newBuilder()</pre> is intended to always create a new instance, not use a cached version.
+ * <p>
+ * If multiple implementations of RestClientBuilder are discovered, the one with the highest value of the {@link Priority} annotation is used
+ * <p>
+ * The {@link ServiceLoader} will first search via the current Thread's Context ClassLoader, then {@link RestClientBuilder}'s {@link ClassLoader}
+ * <p>
+ * The <pre>RestClientBuilder</pre> is a {@link Configurable} class as defined by JAX-RS.  This allows a user to register providers,
+ * implementation specific configuration.
+ */
+public abstract class RestClientBuilder implements Configurable<RestClientBuilder>{
```

Comment:
Is there any relevant reason why RestClientBuilder needs to be an abstract class? It would work equally fine as an interface which wouldn't force inheritance. See #28